File: Go/Dequeue/dequeue_test.go
package Dequeue
import "testing"




func Test_size(t *testing.T) {
    testee := new(Dequeue)
    if testee.size() != 0 {
        t.Errorf("size of empty dequeue should be 0")
    }
    testee.addFront(1)
    if testee.size() != 1 {
        t.Errorf("size after one addFront, should be 1")
    }
}

func Test_addFront(t *testing.T) {
    testee := new(Dequeue)
    testee.addFront(1)
    testee.addFront(2)
    testee.addFront(3)

    if testee.removeRear() != 1 {
        t.Errorf("the first item should be read properly")
    }
}

func Test_addRear(t *testing.T) {
    testee := new(Dequeue)
    testee.addRear(1)
    testee.addRear(2)
    testee.addRear(3)

    if testee.removeRear() != 3 {
        t.Errorf("the last item should be read properly")
    }
}

func Test_removeFront(t *testing.T) {
    testee := new(Dequeue)
    testee.addRear(1)
    testee.addRear(2)
    testee.addRear(3)

    if testee.removeFront() != 1 {
        t.Errorf("the first item should be the same as the one put in")
    }

    if testee.removeFront() != 2 {
        t.Errorf("the second item should be the same as the one put in")
    }

    if testee.removeFront() != 3 {
        t.Errorf("the third item should be the same as the one put in")
    }

    if testee.size() != 0 {
        t.Errorf("after removing all items, the dequeue should have length 1")
    }
}

File: Java/src/Dequeue/DequeueTest.java
package Dequeue;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by Daniel on 4/5/2015.
 */
public class DequeueTest {

    @Test
    public void testSize() throws Exception {
        Dequeue<Integer> testee = new Dequeue<Integer>();
        assertEquals(0, testee.size());

        testee.addFront(1);

        assertEquals(1, testee.size());
    }

    @Test
    public void testAddFront() throws Exception {
        Dequeue<Integer> testee = new Dequeue<Integer>();
        testee.addFront(1);
        testee.addFront(2);
        testee.addFront(3);
        assertEquals(new Integer(1), testee.removeRear());
    }

    @Test
    public void testAddRear() throws Exception {
        Dequeue<Integer> testee = new Dequeue<Integer>();
        testee.addRear(1);
        testee.addRear(2);
        testee.addRear(3);
        assertEquals(new Integer(3), testee.removeRear());
    }

    @Test
    public void testRemoveFront() throws Exception {
        Dequeue<Integer> testee = new Dequeue<Integer>();
        testee.addRear(1);
        testee.addRear(2);
        testee.addRear(3);
        assertEquals(new Integer(1), testee.removeFront());
        assertEquals(new Integer(2), testee.removeFront());
        assertEquals(new Integer(3), testee.removeFront());
        assertEquals(0, testee.size());
    }
}
